spec/stack_master/sparkle_formation/compile_time/allowed_pattern_validator_spec.rb

```diff
@@ -0,0 +1,67 @@
+require_relative '../../../../lib/stack_master/sparkle_formation/compile_time/allowed_pattern_validator'
```

Comment:
This should be in the `spec_helper`

Reply:
I can pull some out..Some fail if I remove the requires. There is something odd with class loading going on in the library with specs

Reply:
They do use autoloading which might interfeer

Reply:
Looks like naming is also not great in the library. Too many of the same names like "SparkleFormation" that are also in other gems

Reply:
naming things the same is fine, as long as you understand the repercussions. Having a top level name of SparkleFormation is a [Monkey Patch](http://culttt.com/2015/06/17/what-is-monkey-patching-in-ruby/). Defining `StackMaster::SparkleFormation` creates a seperate namespace that can be used without interfering with the SparkleFormation namespace.

Using `::SparkleFormation` ensures we are referencing the top level namespace, since by default Ruby will traverse the namespace tree (trying each parent until it finds a match).

[This](https://github.com/envato/stack_master/pull/195/files#diff-aa9e93040542f6109922463ded929141R8) I've never seen before and I'm unsure how this may impact the way libraries are loaded. Can we try without it?